src/sql/workbench/services/query/common/gridDataProvider.ts

```diff
@@ -101,6 +102,7 @@ export async function copySelectionToClipboard(clipboardService: IClipboardServi
 		const eol = provider.getEolString();
 		const valueSeparator = '\t';
 		const shouldRemoveNewLines = provider.shouldRemoveNewLines();
+		const shouldAvoidNewLineAfterTailingLineBreak = provider.shouldAvoidNewLineAfterTailingLineBreak();
```

Comment:
Why is this a provider setting? We control this on the ADS side so this seems like it should just be a general thing for all query editors.

Reply:
Guess I was wrong, since you had to update it here too : https://github.com/microsoft/sqltoolsservice/pull/2122/files

Why do we have logic for copying results in two separate places @alanrenmsft ?

Reply:
Oh, ok nevermind. I didn't realize this was the grid data provider so has nothing to do with the query provider. 

And I just remembered why we have the copy in two places - it's because of the provider-based copy logic @alanrenmsft added. So the implementation in ADS is the "fallback" for providers that didn't implement it. So nevermind for all of this then!